psycop/common/cohort_definition.py

```diff
@@ -52,32 +50,52 @@ def get_outcome_timestamps() -> pl.DataFrame:
         ...
 
 
+msg = Printer(timestamp=True)
+
+
 def filter_prediction_times(
-    prediction_times: pl.DataFrame,
+    prediction_times: pl.LazyFrame,
     filtering_steps: Iterable[PredictionTimeFilter],
     entity_id_col_name: str,
+    get_counts: bool = True,
 ) -> FilteredPredictionTimeBundle:
+    """Apply a series of filters to prediction times.
+
+    If prediction_times is a DataFrame, then the filters will be applied eagerly and we can calculate the stepdeltas.
```

Comment:
but it has to be lazy frame, right? due to the typehint. but get_counts() transforms it into a dataframe